lib/charts/chart.rb

```diff
@@ -17,20 +17,34 @@ def validate_arguments(data, options)
     raise ArgumentError.new('Data missing') if data.empty?
     raise ArgumentError.new('Data not an array') unless data.is_a? Array
     raise ArgumentError.new('Options missing') unless options.is_a? Hash
-    if options[:outer_margin] and !options[:outer_margin].is_a?(Numeric)
-      raise ArgumentError.new('outer_margin not a number')
+    if options[:outer_margin] && !options[:outer_margin].is_a?(Numeric)
+      raise ArgumentError, 'outer_margin not a number'
     end
-    validate_array_and_count(data, options, :colors)
-    validate_array_and_count(data, options, :labels)
+    is_an_array?(options, [:labels, :colors])
+    validate_colors(data, options, :colors)
+    validate_labels(data, options, :labels)
   end
 
-  def validate_array_and_count(data, options, key)
+  def is_an_array?(options, keys)
+    keys.map do |key|
+      if options[key] && !options[key].is_a?(Array)
+        raise ArgumentError, "#{key} not an array"
+      end
+    end
+  end
```

Comment:
I agree the name is not good.  Not sure this name is better... It may be better to simply coerce to an array, and have that raise naturally if this can not be done.